activerecord/lib/active_record/relation/query_methods.rb

```diff
@@ -1042,7 +1042,11 @@ def build_select(arel)
         if select_values.any?
           arel.project(*arel_columns(select_values.uniq))
         else
-          arel.project(@klass.arel_table[Arel.star])
+          if @klass.ignored_columns.any?
```

Comment:
We don't *need* it, but it does make for much nicer looking queries in the common case. 😕